src/Conversion/ONNXToKrnl/Math/Elementwise.cpp

```diff
@@ -1050,11 +1051,32 @@ template <>
 Value emitScalarOpFor<ONNXEqualOp>(ConversionPatternRewriter &rewriter,
     Location loc, Operation *op, Type elementType,
     ArrayRef<Value> scalarOperands) {
+
   CheckIfCustomScalarOpIsSupported<ONNXEqualOp>(elementType);
+  Value results;
   Value lhs = scalarOperands[0];
   Value rhs = scalarOperands[1];
-  MultiDialectBuilder<MathBuilder> create(rewriter, loc);
-  return create.math.eq(lhs, rhs);
+  MultiDialectBuilder<KrnlBuilder, MathBuilder> create(rewriter, loc);
+  Type inputElemType = getElementType(lhs.getType());
+
+  // If the two input values are a string then we want to use the krnlstrncmp.
+  // However, if the input values are a float or an int we can simply use the
+  // equal function.
+  if (inputElemType.isa<krnl::StringType>()) {
+    Value strlenRes = create.krnl.strlen(lhs);
+    Value strncmpRes = create.krnl.strncmp(lhs, rhs, strlenRes);
+    // We need to convert the results to return *i1 since krnlstrncmp returns
+    // i32.
+    Value convertedRes = create.math.cast(rewriter.getI1Type(), strncmpRes);
```

Comment:
this looks redundant, we can create a zeroVal with `strncmpRes.getType()` then compare it to `convertedRes'.

Reply:
You are right... let me fix it. Thanks!!

Reply:
FYI- we do not even need `convertedRes`.... `convertedRes` is `i1` and `strncmpRes.getType()` is `i32`.

Reply:
I have an easy solution.... I changed it to `elementType` which is `i1`.